src/hints/state.rs

```diff
@@ -323,6 +326,50 @@ pub fn enter_scope_commitment_info_by_address(
     Ok(())
 }
 
+pub const WRITE_SPLIT_RESULT: &str = indoc! {r#"
+    from starkware.starknet.core.os.data_availability.bls_utils import split
+
+    segments.write_arg(ids.res.address_, split(ids.value))"#
+};
+pub fn write_split_result(
+    vm: &mut VirtualMachine,
+    _exec_scopes: &mut ExecutionScopes,
+    ids_data: &HashMap<String, HintReference>,
+    ap_tracking: &ApTracking,
+    _constants: &HashMap<String, Felt252>,
+) -> Result<(), HintError> {
+
+    // this hint fills in a Cairo BigInt3 by taking a felt (ids.value) and passing it to a split fn
+    let value = get_integer_from_var_name(vars::ids::VALUE, vm, ids_data, ap_tracking)?;
+    let res_ptr = get_ptr_from_var_name(vars::ids::RES, vm, ids_data, ap_tracking)?;
+
+    fn split(num: Felt252) -> Vec<Felt252> {
```

Comment:
This might belong somewhere else, should we start a module for data availability hints? I'm fine leaving it here for now, too.

Reply:
I think this one is defined in a file called `bls_utils.py`, so you can definitely create a `bls_utils.rs` and put it there.